compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1311,6 +1316,24 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
       else
          generateRegRegInstruction(CMOVERegReg(trueValIs64Bit), node, trueReg, falseReg, cg);
       }
+   else if (conditionOp.isCompareForOrder() && condition->getFirstChild()->getOpCode().isIntegerOrAddress() && !longCompareOn32bit)
```

Comment:
Similarly, it may be worthy bring `!longCompareOn32bit` as the first condition, so that the rest checks can be skipped when `!longCompareOn32bit` is false.

Reply:
addressed.